File: Solutions/Normal/3583. Count Special Triplets/Solution.py
class Solution:
    def specialTriplets(self, nums: List[int]) -> int:
        result: int = 0
        left_handside: dict[int, int] = {}
        right_handside: dict[int, int] = {}
        MOD: int = 1000000007
        for num in nums:
            if (num in right_handside):
                right_handside[num] += 1
            else:
                right_handside[num] = 1
        for num in nums:
            right_handside[num] -= 1
            double_num: int = num * 2
            if (double_num in left_handside):
                result = (result + right_handside[double_num] * left_handside[double_num]) % MOD
            if (num in left_handside):
                left_handside[num] += 1
            else:
                left_handside[num] = 1
        return result

File: Solutions/Normal/3583. Count Special Triplets/Solution.cs
public class Solution {
    public int SpecialTriplets(int[] nums) {
        const int MOD = (int)1e9 + 7;
        long result = 0;
        Dictionary<int, int> rightHandSide = new Dictionary<int, int>(), leftHandSide = new Dictionary<int, int>();
        foreach(int num in nums){
            if (!rightHandSide.ContainsKey(num)){
                rightHandSide[num] = 1;
            }
            else rightHandSide[num]++;
        }
        foreach(int num in nums){
            rightHandSide[num]--;
            int double_num = num * 2;
            if (leftHandSide.ContainsKey(double_num)){
                result = (result + ((long)leftHandSide[double_num] * rightHandSide[double_num])% MOD) % MOD;
            }
            if (!leftHandSide.ContainsKey(num)){
                leftHandSide[num] = 1;
            }
            else{
                leftHandSide[num]++;
            }
        }
        return (int) result;
    }
}